### ISSUE TITLE
Implement 'Flight Phase'

### ISSUE THREAD

**Issue by joshua-jerred** [joshua-jerred/Giraffe#215]
Dependent on #214 

Flight Phases:
- [x] Pre-Launch: Starting mode, ~~(no telemetry?)~~
- [x] Launch: Switch from pre-launch is a manual switch made via a command sent via TCP (Web Client)
- [x] Ascent: The switch from launch to ascent should happen as soon as the flight computer begins to increase in altitude.
- [x] Descent: Dependent on detecting a balloon pop (Separate Issue)
- [x] Recovery: Switch to this mode when the flight computer has landed.

Flight Phase needs to be added to flight_runner_data so that it's immune to hard faults.

The current flight phase also needs to be within shared data

Add the data to the data stream and the server module.
